specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2019-12-01-preview/definitions.json

```diff
@@ -5090,6 +5090,23 @@
           "type": "string",
           "description": "User Password. If no value is provided, a default password is generated."
         },
+        "appType": {
+          "type": "string",
+          "description": "Determines the type of application which send the create user request. Default is old publisher portal.",
+          "enum": [
+            "developerPortal"
+          ],
+          "x-ms-enum": {
+            "name": "AppType",
+            "modelAsString": true,
+            "values": [
+              {
+                "value": "developerPortal",
+                "description": "User create request was sent by new developer portal."
+              }
+            ]
+          }
+        },
```

Comment:
Usually adding new field to the response is a breaking change. 
Check under versioning: [https://armwiki.azurewebsites.net/api_contracts/Intro.html](https://armwiki.azurewebsites.net/api_contracts/Intro.html)

In order to ensure existing clients are not broken,
1. Add "none" enum as well.. 
2. set default to  "none" (for existing resources), while serving GETS and store as None at the next opportunity

Reply:
@sanjaiganesh I have done 2 things to avoid breaking change
1. I have modeled the enum as `string`
2. This property is only needed in the PUT request to create the resource. This property is not returned on `GET` or even response of `PUT`

Reply:
hey Samir - why is it not returned in the response? Unless it is a secret, it should be returned in the response no? Else how will customer update it for example?

Reply:
@solankisamir - one note above

Reply:
@ravbhatnagar This property is used to signify how the user-created be **invited** to join an api program. 
When a user is created, he is sent an email to join the api program. The email contains a website link. Today we have 2 portals, not azure portal, but website we set up for our customer for **his** api program. One of these (website) will actually be deprecated in the future. This `appType` property decides, which link (which website) should we give to the user who is joining our customers api program. 
Ideally, we should have put it in the query param, as it just signifies an action to take on the server-side. But arm template don't work with Query params. Hence this weird semantics.